web/dashboard/src/components/serviceDetail/serviceDetail.jsx

```diff
@@ -175,35 +177,113 @@ class Ingress extends Component {
 }
 
 class Deployment extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isCopied: false,
+    };
+  }
+
+  handleCopyClick() {
+    this.setState({ isCopied: true });
+
+    setTimeout(() => {
+      this.setState({ isCopied: false });
+    }, 2000);
+  };
+
   render() {
-    const { deployment, service, repo, gimletClient, setLogsOverlayVisible, setLogsOverlayNamespace, setLogsOverlayService, scmUrl } = this.props;
+    const { deployment, service, repo, gimletClient, config, setLogsOverlayVisible, setLogsOverlayNamespace, setLogsOverlayService, scmUrl } = this.props;
 
     if (deployment === undefined) {
       return null;
     }
 
+    let hostPort = "<host-port>"
+    let appPort = "<app-port>"
+    if (config) {
+      appPort = config.values.containerPort ?? 80;
```

Comment:
There is one special case that you need to incorporate here:

There is a situation when there is a deployed app, but no matching *file based* gimlet manifest. This happens when we deploy with magic deploy and use a default gimlet manifest. I think you best match this case by checking for the existence of the file based gimlet config.

In those cases leave appPort to be `<app-port>` (and hardcode a host-port to be 8080 maybe). In these cases the documentation advise the user to set the port, since she never set it in gimlet config, and we can't know it.